pcap/pcapio/read.go

```diff
@@ -152,7 +152,7 @@ func (r *PcapReader) Read() ([]byte, BlockType, error) {
 	}
 	caplen := int(r.byteOrder.Uint32(hdr[8:12]))
 	fullLength := int(r.byteOrder.Uint32(hdr[12:16]))
-	if caplen > int(r.snaplen) {
+	if int(r.snaplen) != 0 && caplen > int(r.snaplen) {
```

Comment:
nit: it doesn't look like you need the int() cast just to check against 0?

Reply:
fixed